src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -360,6 +381,127 @@ public function AllChildrenIncludingDeleted()
+        $this->recursivelyCacheDescendants($baseClass, $parentID);
+        $childIDs = [];
+        if (array_key_exists($parentID, Hierarchy::$all_children_child_ids_cache[$baseClass])) {
+            $childIDs = Hierarchy::$all_children_child_ids_cache[$baseClass][$parentID];
+        }
+        $childNodes = [];
+        foreach ($childIDs as $childID) {
+            if (array_key_exists($childID, Hierarchy::$all_children_obj_cache[$baseClass])) {
+                $childNodes[] = Hierarchy::$all_children_obj_cache[$baseClass][$childID];
+            }
+        }
+        $children = ArrayList::create($childNodes);
+        $owner->extend('updateGetAllChildren', $includeDeleted, $children);
+        return $children;
+    }
+
+    /**
+     * Get the base class with the Hierarchy extension of a DataObject
+     * We need to get this instead of just getting the DataObjectSchema::baseDataClass() to ensure that we
+     * can run DB queries that filter on ParentID
+     */
+    public static function getBaseHierarchyClass(DataObject $obj): string
+    {
+        $baseHierarchyClass = $obj->ClassName;
+        $haveFoundTop = false;
+        do {
+            $parentDataClass = get_parent_class($baseHierarchyClass);
+            if (!$parentDataClass
+                || !$parentDataClass::has_extension(Hierarchy::class)
+                || !is_a($parentDataClass, DataObject::class, true)
+            ) {
+                $haveFoundTop = true;
+            } else {
+                $baseHierarchyClass = $parentDataClass;
+            }
+        } while (!$haveFoundTop);
+        return $baseHierarchyClass;
+    }
+
+    /**
+     * Inner recursive method used by getAllChildren
+     * This will use a hierarchical query method where it will initially fetched from the "second level down"
+     * where the ParentID equals the single DataObject at the "top level" of the hierarchy
+     * It will then fetch from the "third level down" in the hierarchy where the ParentID is of the ParentIDs in
+     * the "second level down". In will continue querying like this until the total number of cached objects has
+     * exceeded the configured value `node_threshold_total`, or all object have been fetched
+     */
+    private function recursivelyCacheDescendants(string $baseClass, int|array $idOrIDs): void
+    {
+        if (is_int($idOrIDs)) {
+            $parentIDs = [$idOrIDs];
+        } else {
+            $parentIDs = $idOrIDs;
+        }
+        $parentIDs = array_filter($parentIDs, function ($parentID) use ($baseClass) {
+            return !array_key_exists($parentID, Hierarchy::$all_children_child_ids_cache[$baseClass]);
+        });
+        if (empty($parentIDs)) {
+            return;
+        }
+        $config = $this->getOwner()->config();
+        $count = count(Hierarchy::$all_children_obj_cache[$baseClass]);
+        $threshold = $config->get('node_threshold_total');
+        if ($count >= $threshold) {
+            return;
+        }
```

Comment:
Something about the way you're doing this isn't right. If I set this YAML:

```yaml
---
After: '#cmshierarchy'
---
SilverStripe\ORM\Hierarchy\Hierarchy:
  node_threshold_total: 1
  node_threshold_leaf: 1
```

Without this PR I get the following:
![image](https://github.com/user-attachments/assets/741e3487-209e-4c2b-8b45-cd839bc99991)

With this PR I get no pages at all:
![image](https://github.com/user-attachments/assets/34839d7c-8fb9-4135-8476-bb6ef2694538)

Reply:
Fixed